packages/desktop-client/src/components/mobile/accounts/AccountTransactions.tsx

```diff
@@ -245,38 +247,61 @@ function TransactionListWithPreviews({
   readonly accountName: AccountEntity['name'] | string;
 }) {
   const { t } = useTranslation();
+  const dateFormat = useDateFormat() || 'MM/dd/yyyy';
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  const onRefresh = useCallback(() => {
+    if (accountId) {
+      dispatch(syncAndDownload({ accountId }));
+    }
+  }, [accountId, dispatch]);
+
   const baseTransactionsQuery = useCallback(
     () =>
       queries.transactions(accountId).options({ splits: 'all' }).select('*'),
     [accountId],
   );
 
+  const [showBalances] = useSyncedPref(`show-balances-${accountId}`);
   const [transactionsQuery, setTransactionsQuery] = useState<Query>(
     baseTransactionsQuery(),
   );
   const {
     transactions,
+    runningBalances,
     isLoading,
     reload: reloadTransactions,
     isLoadingMore,
     loadMore: loadMoreTransactions,
   } = useTransactions({
     query: transactionsQuery,
+    options: {
+      calculateRunningBalances: showBalances === 'true',
+    },
   });
 
-  const { previewTransactions } = useAccountPreviewTransactions({
-    accountId: account?.id || '',
+  const { isSearching, search: onSearch } = useTransactionsSearch({
+    updateQuery: setTransactionsQuery,
+    resetQuery: () => setTransactionsQuery(baseTransactionsQuery()),
+    dateFormat,
   });
 
-  const dateFormat = useDateFormat() || 'MM/dd/yyyy';
-  const dispatch = useDispatch();
-  const navigate = useNavigate();
+  const { previewTransactions, runningBalances: previewRunningBalances } =
+    useAccountPreviewTransactions({
+      accountId: account?.id,
+      options: {
```

Comment:
We don't need to pass startingBalance option here since `useAccountPreviewTransactions` already uses account balance as starting balance